### ISSUE TITLE
Stuck at 2.6.6 nothing work with more than 6ghs since then.

### ISSUE THREAD

**Issue by Transisto** [c00w/bitHopper#630]
I have not been able to find a working BH since this issue was suposedly resolved,

 hang for ~1min while ... "database: DB: writing to database"

I'm still at 2.6.6 and now suffer from having an outdated pool config

BUT at least I can have more than 6ghs on it without it refusing more work.

**Comment by c00w** [comment]
We are working on it. That specific bug is fixed in the latest version.
The known critical bugs are:
LP is broken somehow. I.E. it is causing a lot of invalid shares when we get a new block.
Pool burning, namely that you need to setup multiple workers or else pools bounce us because we pull too much work to fast.
Worker interface: For some reason the worker interface only works in chrome.

If you get a bug in the latest version I am happy to fix it. I just need to know about it. Sometimes I think the bug is caused by one thing and it is caused by something else. I'm not perfect and this program is complicated. I'm trying to clean it up.

I wish I could tell you there was an easy fix for the db bug but if you fix that bug a bunch of other hidden bugs appear. Most of those have been fixed in the latest version but it is too complicated to backport.

**Comment by c00w** [comment]
Update:
Lp may be fixed and we are waiting on confirmation. If that is fixed and you setup multiple workers then the latest version should hold up under high loads. If it doesn't I'd love to hear about it.

**Comment by Transisto** [comment]
It seemed good, up to 13ghs, for ~20min

Then I think it hung for ~1 min on this error :  (edit: repetitively)

03:01:29|bitHopper: Exception in wsgi server loop, restarting wsgi in 60 seconds
[Errno 98] Address already in use: ('', 8337)
03:01:30|database: DB: writing to database

Sorry if I'm mixing issues, My patience/curiosity is a little worn out.

**Comment by c00w** [comment]
Did you accidentally start up 2 bitHopper instances? bitHopper is complaining that the address is already registered.

**Comment by Transisto** [comment]
You are right I might have had two BH at once, the new BH is maxed out at ~4.5 - 5.5ghs

All my miners have output like this "
[27/02/2012 03:42:19] Disconnected from server
[27/02/2012 03:42:20] Warning: work queue empty, miner is idle
[27/02/2012 03:42:30] Failed to connect, retrying...
[27/02/2012 03:42:45] Failed to connect, retrying...
[27/02/2012 03:43:10] Failed to connect, retrying...
[27/02/2012 03:43:31] Failed to connect, retrying...
[27/02/2012 03:43:36] Connected to server
[27/02/2012 03:43:46] Disconnected from server
[27/02/2012 03:43:46] Result: 41d2b693 accepted
[27/02/2012 03:43:48] Warning: work queue empty, miner is idle
[27/02/2012 03:43:53] Result: 6dec1db9 rejected
[27/02/2012 03:44:03] Failed to connect, retrying...
[27/02/2012 03:44:13] Failed to connect, retrying...
[27/02/2012 03:44:28] Failed to connect, retrying...
[27/02/2012 03:44:43] Failed to connect, retrying...

In the last 45min it maxed out at 4.6 , average 4ghs,  
So I close it and use 2.6 , within ~2 minute It's at 13ghs /20ghs
(I'm giving the time because it can take ~5 min for my miners to come back from a backup pool)

Is this working for anyone else ? Should I reinstall my OS for it to work ?

Edit : I may have incorrect version of dependencies. At this time, should I expect BH to check for those by itself ?

**Comment by c00w** [comment]
Try installing gevent 1.0b1 instead of 0.13.6. That might speed things up. Was bitHopper noticably freezing? Also did you have multiple worker configured on at least a couple of pools?

**Comment by c00w** [comment]
Also what OS are you using?

**Comment by Transisto** [comment]
Ubuntu ,  how do I install 1.0b1 , is it in the README ?

**Comment by c00w** [comment]
Ubuntu:
wget http://gevent.googlecode.com/files/gevent-1.0b1.tar.gz
tar -xvf gevent-1.0b1.tar.gz
cd gevent-1.0b1
sudo python setup.py install

**Comment by Transisto** [comment]
I've set bitclockers role to info ~5 min ago and I'm still getting ~10  "Lagging. :bitclockers" per second.

10:40:55|bitHopper: Lagging. :bitclockers
10:40:55|bitHopper: Lagging. :bitclockers
10:40:55|bitHopper: Lagging. :bitclockers
10:40:55|bitHopper: Lagging. :bitclockers
10:40:55|bitHopper: Lagging. :bitclockers

**Comment by c00w** [comment]
It may still be in the server map. Restart bitHopper?

**Comment by Transisto** [comment]
http://pastebin.com/41rwRQqv

**Comment by c00w** [comment]
Fix is pushed. Can you tell me if they still appear with the latest version?

**Comment by Transisto** [comment]
it fixed it, thank you, 

sometime I feel like I'm the only one using this software.  ;)

**Comment by Transisto** [comment]
been able to reach 18g ,,, looks good

**Comment by c00w** [comment]
Sweet. And you're not the only person using this software. But a lot of people find a version they like and stick with it. Not a lot of people are willing to test new version etc..

I am extremely grateful. Without testing this software doesn't improve. I do what I can but more eyes always help.

**Comment by Transisto** [comment]
BH died completely,

this : http://pastebin.com/hVmuqjpL

**Comment by c00w** [comment]
Hmm. Your miner started breaking connections. I fixed some behavior when we get getworks we don't have merkle roots for that might fix it. Can you update and tell me if it is still broken?

**Comment by Transisto** [comment]
I didn't wait long to see if it dies,  This new V does not scale past 5ghs

**Comment by c00w** [comment]
The latest version died? Were any trace-backs appearing?

**Comment by Transisto** [comment]
Let me try again, No it didn't die, It just woulnt scale as much as the prev one ,,,

BRB in 15min

**Comment by Transisto** [comment]
http://pastebin.com/sufuLm9w

as said it does not scale past 4-5 ghs, 

I have about 5-6 pool set with one worker each.

**Comment by c00w** [comment]
Add multiple workers. Lots of pools can't serve more than 4-5GH/s of work at the rate we pull it.

**Comment by Transisto** [comment]
Why is it I never had trouble pulling 20ghs before ? ...  Will do if you think that help....

**Comment by c00w** [comment]
Now we pull work more rapidly. So we trip pool rate monitors. Adding multiple workers spreads out the load and causes pools to serve us work faster and not lag out. We've investigated this on other people's clusters and this fixed a lot of issues. They have them scaling to ~40-50GH/s.

**Comment by c00w** [comment]
@Transisto Is this still broken for you?

**Comment by Transisto** [comment]
no it's working fine.
